napari/components/viewer_model.py

```diff
@@ -134,13 +134,15 @@ class ViewerModel(KeymapProvider, MousemapProvider, EventedModel):
 
     # 2-tuple indicating height and width
     _canvas_size: Tuple[int, int] = (600, 800)
-    _ctx: Context
+    _ctx: Mapping
     # To check if mouse is over canvas to avoid race conditions between
     # different events systems
     _mouse_over_canvas: bool = False
 
     def __init__(self, title='napari', ndisplay=2, order=(), axis_labels=()):
         # max_depth=0 means don't look for parent contexts.
+        from .._app.context import create_context
```

Comment:
Similar to comment above - Would there be anyway to avoid this pattern of imports here - could this stuff be passed to the ViewerModel on creation? Could it actually live on the Qt side of the fence? I know this change is just a refactor from what we have right now, but looking at it again makes it jump out to me.

Reply:
yep, agreed.  i don't like it either.

> Could it actually live on the Qt side of the fence

not this though ... it's independent of Qt.  but I can imagine the thing that spins up the app somehow declaring what objects need to have a context.  I _think_ that the dependency injection stuff could work here, but not sure yet